src/fty_mdns_sd_mapping.cc

```diff
@@ -0,0 +1,272 @@
+    =========================================================================
+*/
+
+#include "fty_mdns_sd_classes.h"
+
+#include <cxxtools/jsonserializer.h>
+#include <cxxtools/jsondeserializer.h>
+
+//  --------------------------------------------------------------------------
+// ServiceDeviceMapping
+//
+
+ServiceDeviceMapping::ServiceDeviceMapping():
+    m_serviceHostname(),
+    m_serviceName(),
+    m_serviceAddress(),
+    m_servicePort(),
+    m_extendedInfo()
+{}
+
+ServiceDeviceMapping::ServiceDeviceMapping(std::string hostname, std::string name, std::string address, std::string port, ExtendedInfoMapping extendedInfo) :
+    m_serviceHostname(hostname),
+    m_serviceName(name),
+    m_serviceAddress(address),
+    m_servicePort(port),
+    m_extendedInfo(extendedInfo)
+{}
+
+void ServiceDeviceMapping::fillSerializationInfo(cxxtools::SerializationInfo& si) const
+{
+    try {
+        si.addMember(SERVICE_HOST_NAME_ENTRY) <<= m_serviceHostname;
+        si.addMember(SERVICE_NAME_ENTRY) <<= m_serviceName;
+        si.addMember(SERVICE_ADDRESS_ENTRY) <<= m_serviceAddress;
+        si.addMember(SERVICE_PORT_ENTRY) <<= m_servicePort;
+        si.addMember(SERVICE_EXTENDED_INFO_ENTRY).setCategory(cxxtools::SerializationInfo::Object);
+        cxxtools::SerializationInfo *extended = si.findMember(SERVICE_EXTENDED_INFO_ENTRY);
+        if (extended) {
+            for (auto &info : m_extendedInfo) {
+                extended->addMember(info.first) <<= info.second;
+            }
+        }
+    }
+    catch(const std::exception& e) {
+        throw std::runtime_error("Error in json: " + std::string(e.what()));
+    }
+}
+
+void ServiceDeviceMapping::fromSerializationInfo(const cxxtools::SerializationInfo& si)
+{
+    try {
+        si.getMember(SERVICE_HOST_NAME_ENTRY) >>= m_serviceHostname;
+        si.getMember(SERVICE_NAME_ENTRY) >>= m_serviceName;
+        si.getMember(SERVICE_ADDRESS_ENTRY) >>= m_serviceAddress;
+        si.getMember(SERVICE_PORT_ENTRY) >>= m_servicePort;
+        auto &extended = si.getMember(SERVICE_EXTENDED_INFO_ENTRY);
+        for (auto &member : extended) {
+            std::string key = member.name();
+            std::string value;
+            member.getValue(value);
+            m_extendedInfo.insert(make_pair(key, value));
+        }
+    }
+    catch(const std::exception& e) {
+        throw std::runtime_error("Error in json: " + std::string(e.what()));
+    }
+}
+
+std::string ServiceDeviceMapping::toString() const
+{
+    std::string returnData("");
+
+    try {
+        cxxtools::SerializationInfo si;
+        fillSerializationInfo(si);
+
+        std::stringstream output;
+        cxxtools::JsonSerializer serializer(output);
+        serializer.beautify(true);
+        serializer.serialize(si);
+        returnData = output.str();
+    }
+    catch(const std::exception& e) {
+        throw std::runtime_error("Error in json: " + std::string(e.what()));
+    }
```

Comment:
What could throw an exception during serialization? Consider removing the try/catch block.